### PR TITLE
PDCpy enhancement

### PR DESCRIPTION
**Pull request by zhangwei217245**
1. Positional args for check_type to support multiple types
2. Avoid conflicting between multiple ranks for ctrace
3. disable_ctrace
4. add bytes, list, dict, pickle support for metadata attributes

### CODE REVIEW

File: pdc/object.pyx [hunk 1]
```diff
@@ -378,12 +381,15 @@ class Object:
...
                 else:
+                    # sizes, offsets = remoteRegion._get_sizes_offsets(object.dims)
+                    # region_id, sizes = remoteRegion._construct_with(sizes)
                     out = np.empty(sizes, dtype=object.type.as_numpy_type())
                     transfer_id = cpdc.PDCregion_transfer_create(<void *> <size_t> out.ctypes.data, type(self).RequestType.GET.value, object._id, local_region_id, region_id)
                     ctrace('region_transfer_create', transfer_id, out, type(self).RequestType.GET, object._id, local_region_id, region_id)
                     if transfer_id == 0:
                         raise PDCError('failed to create transfer')
                     self._id = transfer_id
+                    print(f"shape of out: {out.shape}")
```
Comment (houjun) [inline pdc/object.pyx:392]: Can you comment out this debug print?

File: pdc/object.pyx [hunk 2]
```diff
@@ -346,6 +346,8 @@ class Object:
             checktype(object, 'object', Object)
             checktype(request_type, 'request type', type(self).RequestType)
             region_id, sizes = remoteRegion._construct_with(object.dims)
+            print(f"sizes: {sizes}, object.dims: {object.dims}")
```
Comment (houjun) [inline pdc/object.pyx:349]: Can you comment out this debug print?

**Review by houjun — CHANGES_REQUESTED** [review]

File: pdc/object.pyx (hunk 1 above)
Comment (zhangwei217245) [reply pdc/object.pyx]: fixed.

File: pdc/object.pyx (hunk 2 above)
Comment (zhangwei217245) [reply pdc/object.pyx]: fixed.

**Comment by zhangwei217245** [comment]
I would like to remind that the aforementioned debugging logs are commented out. we can go ahead and merge the PR.

**Review by houjun — APPROVED** [review]

**Pull request merged**